lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -646,6 +646,54 @@ OpFoldResult AtenToDtypeOp::fold(ArrayRef<Attribute> operands) {
   return getOperand(0);
 }
 
+//===----------------------------------------------------------------------===//
+// AtenToDtypeLayoutOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenToDtypeLayoutOp::fold(ArrayRef<Attribute> operands) {
+  bool nonBlocking, copyArg, pinMemory;
+  // The pin_memory arg must be `False` or `none`.
+  if (!memory_format().getType().isa<Torch::NoneType>() &&
+      (!matchPattern(pin_memory(), m_TorchConstantBool(&pinMemory)) ||
+       pinMemory))
+    return nullptr;
+  // The non_blocking arg must be `False`.
+  if (!matchPattern(non_blocking(), m_TorchConstantBool(&nonBlocking)) ||
```

Comment:
similar to the above. `nonBlocking` could be a variable and `nonBlocking` should only be checked when it's a constant bool.